### PR TITLE
Rename authentication method

### PR DESCRIPTION
**Pull request by 3lvis**
# Deprecates
```swift
public func authenticate(username: String, password: String)
public func authenticate(token: String)
public func authenticate(headerKey: String = "Authorization", headerValue: String)
```

# Adds
```swift
public func setAuthorizationHeader(username: String, password: String)
public func setAuthorizationHeader(token: String)
public func setAuthorizationHeader(headerKey: String = "Authorization", headerValue: String)
```

### CODE REVIEW

File: Sources/Networking.swift [hunk 1]
```diff
@@ -154,37 +152,60 @@ public class Networking {
...
-            self.session = URLSession(configuration: config)
-        }
+        self.basicAuthenticationHeaderField = (username, password)
     }
 
     /** 
      Authenticates using a Bearer token, sets the Authorization header to "Bearer \(token)".
      - parameter token: The token to be used.
      */
+    @available(*, deprecated: 2.2.0, message: "Use `var authenticationHeaderFieldValue` with \"Bearer \\(token)\" as value instead.")
```
Comment (Elland) [inline Sources/Networking.swift]: Why? This seems too error prone. Can we make this take only a token?

File: Tests/POSTTests.swift [hunk 2]
```diff
@@ -219,7 +219,7 @@ class POSTTests: XCTestCase {
     }
 
     func deleteAllCloudinaryPhotos(networking: Networking, cloudName: String, secret: String, APIKey: String) {
-        networking.authenticate(username: APIKey, password: secret)
+        networking.basicAuthenticationHeaderField = (APIKey, secret)
```
Comment (Elland) [inline Tests/POSTTests.swift]: What is a a Cloudinary Photo?

Comment (3lvis) [inline Tests/POSTTests.swift]: Unit tests for multipart.

File: Sources/Networking.swift (hunk 1 above)
Comment (3lvis) [inline Sources/Networking.swift]: I don't follow.

Comment (3lvis) [inline Sources/Networking.swift]: Provide a different variable to just set the token?

Comment (Elland) [inline Sources/Networking.swift]: Why is this being set directly? The methods I had before worked better than this. 😢 Less error prone, more clarity, no need to rely on tuples...

File: Tests/POSTTests.swift (hunk 2 above)
Comment (Elland) [inline Tests/POSTTests.swift]: That's a hell of a name. 😂

**Comment by 3lvis** [comment]
This is most likely the best approach so far. Consistent and correct.

**Comment by Elland** [comment]
Much better. 👍

**Pull request merged**